### DISCUSSION TITLE
Authentication failed:Does BMW add graphic varification code verify?

### DISCUSSION THREAD

**Discussion by ouhaiorg** [bimmerconnected/bimmer_connected#415] [Q&A] [answered]
bash-5.1# bimmerconnected fingerprint *** *** china               DEBUG:bimmer_connected.account:Getting vehicle list
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): myprofile.bmw.com.cn:443
DEBUG:urllib3.connectionpool:https://myprofile.bmw.com.cn:443 "GET /eadrax-coas/v1/cop/publickey HTTP/1.1" 200 356
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): myprofile.bmw.com.cn:443
DEBUG:urllib3.connectionpool:https://myprofile.bmw.com.cn:443 "POST /eadrax-coas/v1/login/pwd HTTP/1.1" 422 111
ERROR:bimmer_connected.account:Authentication failed (True): 该账户需图形验证码校验！
Translation to ENGLISH:this account need graphic verification code to  verify.

**Comment by ouhaiorg** [comment]
my bimmer_connected is 0.8.10
these days ago using version 0.8.7 always authentication success.

**Reply by ouhaiorg** [reply]
Because the 0.8.7 can't work suddenly, then I upgrade to 0.8.10.

**Comment by rikroe** [comment]
So 0.8.10 doesn't work, but 0.8.7 or 0.8.9 do?
I will try to log the log in process again and see what happens.

What happens if you log in via the My BMW app using password (not SMS login)?

**Reply by ouhaiorg** [reply]
![Screenshot_2022-01-16-20-01-56-99_2769d0ca37355d194b0fe3f4241b60fe](https://user-images.githubusercontent.com/29619015/149659227-5eee2a6a-f5a0-43c9-9d35-6ad38c5d195b.jpg)

**Reply by ouhaiorg** [reply]
My BMW app add a graphic varification really!

**Reply by rikroe** [reply]
Oh. I will have a look, but no promise that we will get it working.

**Reply by ouhaiorg** [reply]
Thanks, Now  the bimmer_connected is back to normal, I just according to opinion of @muxiachuixue, logout MyBMW app and login again.

**Comment by muxiachuixue** [comment, marked as answer]
This is because the user has entered the wrong password for too many times. Try to open MyBMW app to logout and login again. Then this error will disappear.

**Reply by ouhaiorg** [reply]
![IMG_20220117_113709](https://user-images.githubusercontent.com/29619015/149704815-ecabc7be-0a5b-4538-b239-eafd6a4e43cd.jpg)
你好，很高兴认识你，我试过注销MyBMW应用，重新登陆，还是需要图形验证码安全验证，我觉得BMW中国应该新增加了这种安全验证，要不你用MyBMW试试看。我的账号密码一直存在homeassistant配置文件里，看记录1月12日还能用，之后就登陆出错了。

**Reply by muxiachuixue** [reply]
> 
我的正常的，你说的情况我之前遇到过，MyBMW 注销后重新登录，然后输入验证码，就好了。建议你多注销登录几次，如果还不行，可以尝试更改一下密码。

**Reply by ouhaiorg** [reply]
![IMG_20220117_113749](https://user-images.githubusercontent.com/29619015/149733498-356861b2-8c29-41b6-b20d-8551db49f35b.jpg)
现在HA里用bimmer_connected插件可以正常使用了，我因为看到MyBMW第一次出现了图形验证，以为bimmer_connected里也要图形验证，所以后来就没在HA里测试了。非常感谢！
另外我还在车里接了gt06，弄了个gt06服务器和HA插件，这样可以实时得到汽车的位置、速度、方向等，有机会一起研究。

**Comment by mjxoicq** [comment]
My bimmer connected also stopped working several days ago. I guess maybe it is the same problem?

Logger: homeassistant.config_entries
Source: config_entries.py:344
First occurred: 15:18:41 (1 occurrences)
Last logged: 15:18:41

Config entry 'BMW' for bmw_connected_drive integration not ready yet: 403 Client Error: Forbidden for url: https://myprofile.bmw.com.cn/eadrax-coas/v1/cop/publickey; Retrying in backgroun

**Reply by muxiachuixue** [reply]
Please upgrade to [0.8.10](https://github.com/bimmerconnected/bimmer_connected/releases/tag/0.8.10)

**Reply by mjxoicq** [reply]
不好意思，我想请教一下怎么升级。把“bimmer_connected-0.8.10”下载下来，解压到哪个文件夹下？我在ha的目录里没找到类似的

**Reply by ouhaiorg** [reply]
not need download bimmer_connected-0.8.10, just download ha_custom_component, unzip bmw_connected_driver on custom_component directory, and edit file custom_components/bmw_connected_drive/manifest.json, change "requirements": ["bimmer_connected==0.8.7"] to ==0.8.10, restart homeassistant

**Reply by mjxoicq** [reply]
Thank you so much! It's working for now.

**Comment by rikroe** [comment]
You can now also use the latest custom component directly: https://github.com/bimmerconnected/ha_custom_component/releases/tag/20220117.1
